src/ll.GoogleTranslator.js

```diff
@@ -0,0 +1,131 @@
+/*!
+ * LL two-way parallel translation - Google translator class
+ *
+ * @copyright 2019 LL team and others; see LICENSE.txt for terms
+ */
+
+/**
+ * Google translator
+ * @abstract
+ * @class
+ *
+ * @constructor
+ * @param {Object} config
+ * @param {string} config.key API key
+ */
+ll.GoogleTranslator = function LLGoogleTranslator( config ) {
+	this.url = 'https://translation.googleapis.com';
+	if ( !config.key ) {
+		throw new Error( 'Need Google API key' );
+	}
+	this.key = config.key;
+	// Parent constructor: must be called after setting .url and .key above
+	ll.GoogleTranslator.super.call( this );
+};
+
+/* Initialization */
+
+OO.inheritClass( ll.GoogleTranslator, ll.Translator );
+
+// TODO: Make separators more robust.
+// Google messes with multi-character separators, usually by adding spaces betweem unalike character
+// repetitions, such as ":!!:" becoming ": !!:". Using more obscure characters works, but is still
+// inherently fragile.
+ll.GoogleTranslator.static.outerSeparator = '☆';
```

Comment:
To reduce the likelihood of the separator occurring organically in the text, you can use a sequence of emoji, for example I tested with '🍇🍈🍉🍊🍋🍌'.

I found that spaces were only added to recognisable punctuation, so this didn't suffer from the same whitespace issues.

Reply:
Looks like with Google Translate you can directly pass an array of strings, meaning we probably don't need separators in this case.

Reply:
I'll leave this for now since we're going to switch to using an array of strings instead (at least for Google)